nim/tests/test_routes.py

```diff
@@ -0,0 +1,110 @@
+import json
+from nim.app import app
+import pytest
+
+
+@pytest.fixture
+def client_app():
+    return app.test_client()
+
+
+def assert_new_game_success(passed_client, req):
+    """
+    Asserts post to /new is 200
+    & state is not empty
+    :param passed_client: test_client
+    :param req: json to post
+    :return: bool
+    """
+    response = passed_client.post(
+        '/new',
+        content_type='application/json',
+        data=req
+    )
+    assert response.status_code == 200 and not len(json.loads(response.data)['state']) == 0
+
+
+def assert_new_game_failed(passed_client, req):
+    """
+    Asserts post tp /new gets error code 400
+     & the string 'error' exists in the response
+    :param passed_client: test_client
+    :param req: json to post
+    :return: bool
+    """
+    response = passed_client.post(
+        '/new',
+        content_type='application/json',
+        data=req
+    )
+    assert response.status_code == 400 and b'error' in response.data
+
+
+def test_index(client_app):
+    response = client_app.get('/', content_type='html/text')
+    assert response.status_code == 200
+
+
+def test_new_success(client_app):
+    assert_new_game_success(client_app,
+                            json.dumps({'min': 5, 'max': 15, 'piles': 4}))
+
+
+def test_new_omit_min_success(client_app):
+    assert_new_game_success(client_app,
+                            json.dumps({'max': 15, 'piles': 4}))
+
+
+def test_new_omit_max_success(client_app):
+    assert_new_game_success(client_app,
+                            json.dumps({'min': 5, 'piles': 4}))
+
+
+def test_new_omit_piles_success(client_app):
+    assert_new_game_success(client_app,
+                            json.dumps({'min': 5, 'max': 15}))
+
+
+def test_new_omit_all_success(client_app):
+    assert_new_game_success(client_app,
+                            json.dumps({}))
+
+
+def test_new_max_limit_fail(client_app):
+    assert_new_game_failed(client_app,
+                           json.dumps({'min': 10, 'max': 51, 'piles': 3}))
```

Comment:
Since you've shown you can omit some arguments, may be easier to only pass the faulty argument in this function and below.
For example:
```
def test_new_max_limit_fail(client_app):
    assert_new_game_failed(client_app, json.dumps({'max': 51}))
```
Just a thought.

Reply:
And cover both cases at once? 
My thinking was to deal with one odd case at a time, and not introduce multiple failure points. 
But I see what you're saying.. 
I was concerned that if we do it this way, the ability to only pass one param *OR* the limit itself breaking would cause this test to fail

Reply:
But you already tested above with the `_success` functions that passing a subset of the params work, so here we would now be testing bad values for params.

Reply:
Seems like you've missed this comment.
When testing for failure cases, you should only pass what is needed for the test to fail.
For example, if passing `piles=0` should fail, you should NOT also pass `min` and `max`, otherwise we won't know for sure what caused the failure.